tests/purs/failing/CoercibleRoleMismatch1.out

```diff
@@ -1,14 +1,14 @@
 Error found:
 in module [33mMain[0m
-at tests/purs/failing/CoercibleRoleMismatch1.purs:4:1 - 4:29 (line 4, column 1 - line 4, column 29)
+at tests/purs/failing/CoercibleRoleMismatch1.purs:6:1 - 6:27 (line 6, column 1 - line 6, column 27)
 
   Role mismatch for the type parameter [33ma[0m:
 
     The annotation says [33mphantom[0m but the role [33mrepresentational[0m is required.
 
 
 in role declaration for [33mIdentity[0m
-in type constructor [33mIdentity[0m
+in data binding group Identity
```

Comment:
Having foreign data type declarations depend on their role declaration in the binding group desugaring step would complicate self cycles detection, but this leaks in error messages since we do display that role declarations of data types belong to a binding group here. Is there a reason for not eliminating the hint for singleton binding groups?

Reply:
I might even go further and suggest that it should be eliminated whenever another hint narrows down the error to one declaration. Maybe there should be a hint category for top-level declarations or something?

Reply:
I introduced a category for declaration hints.